webpack/assets/javascripts/react_app/redux/index.js

```diff
@@ -1,21 +1,31 @@
 import createLogger from 'redux-logger';
 import thunk from 'redux-thunk';
 import { applyMiddleware, createStore } from 'redux';
-
-import reducer from './reducers';
+import combineReducersAsync from './reducers';
 
 let middleware = [thunk];
 
 if (process.env.NODE_ENV !== 'production' && !global.__testing__) {
   middleware = [...middleware, createLogger()];
 }
 
-const _getStore = () => createStore(
-  reducer,
-  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
-  applyMiddleware(...middleware),
-);
+function configureStore() {
+  const store = createStore(
+    combineReducersAsync(),
+    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
+    applyMiddleware(...middleware),
+  );
+
+  store.asyncReducers = {};
+  return store;
+}
+
+export function registerReducer(store, name, asyncReducer) {
```

Comment:
number of arguments do not correspond to what is in the [example gist](https://gist.github.com/danseethaler/d745921bebbe5f696c4a316e38d8879f)

Worked as expected when I changed it to:

```js
const store = configureStore();

export function registerReducer(name, asyncReducer) {
// no changes here
}

export default store;
```

Reply:
Correct, I refactored a bit to remove the need to expose the store. It should be a smaller api now with just the name and the reducer.